collector.go

```diff
@@ -77,3 +80,37 @@ func execute(log *zap.SugaredLogger, c Collector, ch chan<- prometheus.Metric, o
 	}
 	ch <- prometheus.MustNewConstMetric(scrapeSuccessMetric.desc, scrapeSuccessMetric.valueType, success, c.Name())
 }
+
+func GetEnabledMetricsCollectors(
+	log *zap.SugaredLogger,
+	disabled []configondatv1.MetricsExporterCollector,
+) []Collector {
+	sort.Slice(disabled, func(i, j int) bool {
+		return string(disabled[i]) < string(disabled[j])
+	})
+
+	var metricsCollectors []Collector
+	for name, collectorFactory := range map[configondatv1.MetricsExporterCollector](func() Collector){
+		configondatv1.MetricsExporterCollectorDiskStats:  func() Collector { return NewDiskStatsCollector() },
+		configondatv1.MetricsExporterCollectorFileSystem: func() Collector { return NewFileSystemCollector() },
+	} {
+		if IsCollectorDisabled(disabled, name) {
+			log.Infof("disabling %s collector", name)
+			continue
+		}
+		metricsCollectors = append(metricsCollectors, collectorFactory())
+	}
+	return metricsCollectors
+}
+
+// isCollectorDisabled uses binary search on the list of disabled collector names to determine whether the
+// given collector is disabled. The disabled list must be sorted.
+func IsCollectorDisabled(
+	disabled []configondatv1.MetricsExporterCollector,
+	collector configondatv1.MetricsExporterCollector,
+) bool {
+	i := sort.Search(len(disabled), func(i int) bool {
```

Comment:
Open for discussion but we currently only ever support 2 entries here. Even k8s and node_exporter support like 6 (I think)...
Is it worth using a binary search in place of a simpler for loop that everyone understands and can modify easily?
The gains from the binary search are prob so small we wouldn't even notice